src/main/java/com/ocient/jdbc/XGConnection.java

```
@@ -1600,12 +1600,16 @@ private void tryConfigureSocketOptions(Socket sock){
 				sock.setOption(entry.getKey(), entry.getValue());
 			} catch (IOException | UnsupportedOperationException | IllegalArgumentException | SecurityException ex) {
 				LOGGER.log(Level.WARNING, String.format("Failed setting socket option with message %s."), ex.getMessage());
+			} catch (NoSuchMethodError error){
```

I thought that the exception that was thrown was not one of these but was like no such field or something.

---

I think we really could just catch(Exception e) here

---

catching error won't catch exception: https://stackoverflow.com/questions/38283476/java-nosuchmethoderror-not-caught-by-exception. But I could change the above to catch exception instead of all of those other things.

---

Yes I changed it to just catch all Exceptions in the above line. I don't want to end up in another situation with an exception we didn't cover.